node/toolrunner.ts

```diff
@@ -62,6 +61,18 @@ export interface IExecSyncResult {
     error: Error;
 }
 
+export class Deferred<T> {
+    promise: Promise<T>;
+    resolve: (value: T) => void;
+    reject: (reason?: any) => void;
+    constructor() {
+        this.promise = new Promise((resolve, reject) => {
+            this.resolve = resolve;
+            this.reject = reject;
+        });
+    }
+}
```

Comment:
@JamieMagee I introduced this class because the toolrunner tests started consistently failing after the migration to native promises. I believe it was because the promises were resolving before the expected output was emitted. So, I decided to make the usage of promises more similar to the way it was handled in the Q library, and it fixed the tests.

Reply:
Ok, I got rid of the wrapper class and started using just promise constructors to imitate the deferral. I believe with this approach we can have the same behavior as previously, with all tests passing, while still using promises in a standard way. What do you think?

Reply:
Thank you! This is great ❤️ 

Sorry for being pushy on this issue, but I've got burned before with nonstandard implementations of standard library functions.